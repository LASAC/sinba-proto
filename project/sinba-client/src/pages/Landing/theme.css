body {
  background-color: white;
  background-image: url('/logo.png');
  background-size: 400px;
  background-position: center;
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-position: 50% 30%;
  color: #636b6f;
  font-family: 'Raleway', sans-serif;
  font-weight: 100;
  height: 100vh;
  margin: 0;
}

header h1 {
  color: var(--main-color);
  display: flex;
  font-family: var(--main-font-family);
  font-size: 84px;
  font-weight: 100;
  justify-content: center;
  padding-top: 100px;
}

img {
  width: 350px;
}

.landing {
  font-size: 12px;
}

.landing nav {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding-top: 13px;
}

.landing nav a {
  text-decoration: none;
  color: var(--main-color);
  font-family: var(--main-font-family);
  font-weight: bold;
  letter-spacing: .1rem;
  padding: 5px 10px;
  text-transform: uppercase;
  text-decoration: none;
}

@media screen and (min-width: 501px) /* Tablet */ {
  body {
    background-position: 50% 65%;
  }

  header h1 {
    padding-top: 200px;
  }

  .landing .topNav {
    justify-content: flex-end
  }

  .landing nav a {
    padding: 0 25px;
  }
}

@media screen and (min-width: 980px) /* Desktop */ {

}