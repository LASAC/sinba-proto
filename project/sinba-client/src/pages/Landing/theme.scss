.landing {
  background-color: white;
  background-image: url('/logo.png');
  background-size: 400px;
  background-position: center;
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-position: 50% 100%;
  color: #636b6f;
  font-family: 'Raleway', sans-serif;
  font-size: 12px;
  font-weight: 100;
  height: 100vh;
  margin: 0;

  header {
    h1 {
      color: $main-color;
      display: flex;
      font-family: $main-font-family;
      font-size: 84px;
      font-weight: 100;
      justify-content: center;
      padding-top: 150px;
    }
  }

  img {
    width: 350px;
  }

  nav {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    padding-top: 13px;

    a {
      text-decoration: none;
      color: $main-color !important;
      font-family: $main-font-family;
      font-weight: bold;
      letter-spacing: .1rem;
      padding: 5px 10px;
      text-transform: uppercase;
    }

    .link {
      color: $main-color !important;
      letter-spacing: .1rem;
      font-family: $main-font-family;
      font-size: 12px;
      font-weight: bold;
      padding: 0;
      text-transform: uppercase;
      text-decoration: none;

      button {
        margin: -7px 0;
      }
    }
  }
}

.loginDialog {
  min-width: 300px;
}

@media screen and (min-width: 501px) /* Tablet and Larger */ {
  .landing {
    background-position: 50% 65%;

    header h1 {
      padding-top: 200px;
    }

    nav {
      
      a {
        padding: 0 25px;
      }
      .link {
        padding: 0 30px 0 10px;

        button {
          margin: -11px 0;
        }
      }
    }

    .topNav {
      justify-content: flex-end
    }
  }
}