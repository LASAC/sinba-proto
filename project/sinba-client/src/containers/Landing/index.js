import React from 'react'
// import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
// import { Button } from 'react-toolbox/lib/button'
import { getVersion } from '../../reducers/api-status/actions'
import { login } from '../../reducers/auth/actions'
import Dialog from 'react-toolbox/lib/dialog'
import Login from '../../components/Login'
import strings from '../../services/localization'
import theme from './theme.scss'
import AppBar from '../../components/AppBar'

class Landing extends React.Component {
  state = {
    loginDialogActive: false
  }

  render() {
    const signInButton = {
      to: '/',
      label: strings.signIn,
      onClick: this.toggleLogin
    }
    return (
      <article className={theme.landing}>
        <AppBar leftButtons={[signInButton]} hide={['/']} />
        <header>
          <h1>SINBA</h1>
        </header>
        <main className={theme.main}>
          <nav>
            <a href="https://github.com/LASAC/SINBA/wiki">{strings.documentation}</a>
            <a href="http://lasac.ledes.net">LASAC</a>
            <a href="http://www.ledes.net">LEDES</a>
            <a href="https://ufms.br">UFMS</a>
          </nav>
        </main>
        <div className={theme.footer}>{this.getVersionNote()}</div>
        {this.renderDialog()}
      </article>
    )
  }

  renderDialog = () => {
    const { loginDialogActive } = this.state
    return (
      <Dialog
        actions={this.actions}
        active={loginDialogActive}
        className={theme.loginDialog}
        onEscKeyDown={this.toggleLogin}
        onOverlayClick={this.toggleLogin}
      >
        <Login />
      </Dialog>
    )
  }

  toggleLogin = () => {
    const { loginDialogActive } = this.state
    this.setState({ loginDialogActive: !loginDialogActive })
  }

  getVersionNote = () => {
    const { clientVersion, serverVersion } = this.props
    if (!serverVersion) {
      this.props.getVersion()
      return ''
    }

    return `server: ${serverVersion} | client: ${clientVersion}`
  }
}

const mapStateToProps = (state) => {
  const { clientVersion, serverVersion } = state.apiStatus
  return { clientVersion, serverVersion }
}
const mapDispatchToProps = (dispatch) => bindActionCreators({ getVersion, login }, dispatch)
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Landing)
