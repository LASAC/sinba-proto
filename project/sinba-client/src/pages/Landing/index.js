import React from 'react'
import strings from '../../localization'
import theme from './theme.css'

export default () =>
  <article className={theme.landing}>
    <nav className={theme.topNav}>
      <a href='/login'>{strings.signIn}</a>
      <a href='/register'>{strings.register}</a>
    </nav>
    <header>
      <h1>SINBA</h1>
    </header>
    <main>
      <nav>
        <a href='https://github.com/LASAC/SINBA/wiki'>{strings.documentation}</a>
        <a href='http://lasac.ledes.net'>LASAC</a>
        <a href='http://www.ledes.net'>LEDES</a>
        <a href='https://ufms.br'>UFMS</a>
      </nav>
    </main>
  </article>