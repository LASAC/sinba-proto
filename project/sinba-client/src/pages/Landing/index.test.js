import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Landing from '.'

describe('<Landing />', () => {
  it('should render correctly', () => {
    expect(toJson(shallow(<Landing />))).toMatchSnapshot()
  })
})