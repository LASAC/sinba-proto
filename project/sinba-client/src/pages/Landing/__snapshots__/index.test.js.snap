// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Landing /> should render correctly 1`] = `
<article>
  <nav>
    <a
      href="/login"
    >
      Sign In
    </a>
    <a
      href="/register"
    >
      Register
    </a>
  </nav>
  <header>
    <h1>
      SINBA
    </h1>
  </header>
  <main>
    <nav>
      <a
        href="https://github.com/LASAC/SINBA/wiki"
      >
        Documentation
      </a>
      <a
        href="http://lasac.ledes.net"
      >
        LASAC
      </a>
      <a
        href="http://www.ledes.net"
      >
        LEDES
      </a>
      <a
        href="https://ufms.br"
      >
        UFMS
      </a>
    </nav>
  </main>
</article>
`;
